### ISSUE TITLE
In the JWT token I have set moderator:false but after I joined the meet via token , showing the message that you`re joined as a moderator. Please help me to sort this issue. I have self hosted Jitsi in AWS EC2 Ubuntu 22.04

### ISSUE THREAD

**Issue by karthikeyanreactdev** [jitsi-contrib/prosody-plugins#29]
I edited /etc/prosody/conf.avail/xxxx.xxxx.xxxx.cfg.lua

VirtualHost "xxx.xxxx.xxxx"
authentication = "token" -- do not delete me
-- Properties below are modified by jitsi-meet-tokens package config
-- and authentication above is switched to "token"
app_id="xxxxx"
app_secret="xxxxx"
allow_empty_token = false;
-- Assign this host a certificate for TLS, otherwise it would use the one
-- set in the global section (if any).
-- Note that old-style SSL on port 5223 only supports one certificate, and will always
-- use the global one.

Component "conference.xxx.xxxxxx.xx" "muc"
restrict_room_creation = true
storage = "memory"
modules_enabled = {
"muc_hide_all";
"muc_meeting_id";
"muc_domain_mapper";
"polls";
"token_verification";
"muc_rate_limit";
"muc_password_whitelist";
"muc_allowners";
"token_moderation";
}
admins = { "[focus@auth.xxx.xxxxx.xx](mailto:focus@auth.xxx.xxxxx.xx)" }
muc_password_whitelist = {
"[focus@auth.xxx.xxxx.xx](mailto:focus@auth.xxx.xxxx.xx)"
}
muc_room_locking = false
allowners_moderated_subdomains = { "moderated"}
muc_room_default_public_jids = true

**Comment by damencho** [comment]
> "muc_allowners";

Disable that module.

**Comment by karthikeyanreactdev** [comment]
Its not working

**Comment by karthikeyanreactdev** [comment]
@damencho Please support us to fix this issue

**Comment by damencho** [comment]
Have you restarted prosody?

**Comment by karthikeyanreactdev** [comment]
Yes. But members still joining as moderators

**Comment by emrahcom** [comment]
Looks like you use `token_moderation` to manage roles. The related repo for this module is [this one](https://github.com/nvonahsen/jitsi-token-moderation-plugin). Did you check the instructions in this repo?

**Comment by karthikeyanreactdev** [comment]
yes @emrahcom i have followed all the instructions. but im still facing the issue

**Comment by damencho** [comment]
Share your jicofo config

**Comment by karthikeyanreactdev** [comment]
# Jicofo HOCON configuration. See reference.conf in /usr/share/jicofo/jicofo.jar for
#available options, syntax, and default values.
jicofo {
  xmpp: {
    client: {
      client-proxy: "focus.light.codegroot.in"
      xmpp-domain: "light.codegroot.in"
      domain: "auth.light.codegroot.in"
      username: "focus"
      password: "SYQfnkgbx7WMvaye"
    }
    trusted-domains: [ "recorder.light.codegroot.in" ]
  }
  bridge: {
    brewery-jid: "JvbBrewery@internal.auth.light.codegroot.in"
  }
  conference: {
    enable-auto-owner: false
  }
}

**Comment by damencho** [comment]
And no sip-communicator.properties?

**Comment by karthikeyanreactdev** [comment]
Can i share sip-communicator.properties

**Comment by damencho** [comment]
Mask passwords and share

**Comment by emrahcom** [comment]
Could you share also the content of a token you created?

**Comment by karthikeyanreactdev** [comment]
<img width="960" alt="sip" src="https://github.com/jitsi-contrib/prosody-plugins/assets/81141000/4ba532e2-efd0-4872-bf82-85b2252f41ac">

**Comment by damencho** [comment]
Remove that line and restart jicofo. Does that change anything?

**Comment by karthikeyanreactdev** [comment]
eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb250ZXh0Ijp7InVzZXIiOnsiYXZhdGFyIjoiaHR0cHM6L2dyYXZhdGFyLmNvbS9hdmF0YXIvYWJjMTIzIiwibmFtZSI6IkthcnRoaWtleWFuIiwiZW1haWwiOiJka0B5b3BtYWlsLmNvbSJ9fSwiYXVkIjoiaml0c2kiLCJpc3MiOiJncm9vdGxpZ2h0Iiwic3ViIjoibGlnaHQuY29kZWdyb290LmluIiwicm9vbSI6IioiLCJtb2RlcmF0b3IiOmZhbHNlLCJuYmYiOjE3MDAwMDY5MjMsImV4cCI6MTcxMDAwNjkyM30.-FgdWVC5jPZi9VKf7UWNhyLUEQ_7-_Ehs63myqI3L-g

**Comment by karthikeyanreactdev** [comment]
@damencho after removing that line in sip-communicator.properties I'm unable to join via jwt token

**Comment by damencho** [comment]
That is strange. That should not be used, that tells jicofo that anyone that joins that domain is authenticated and moderator. 
Why it does not join? What is the error? What is prosody log for that?

**Comment by karthikeyanreactdev** [comment]
@damencho now all member joining are not participants even if moderator is true in token

**Comment by damencho** [comment]
That means you need to fix now the module that sets the moderator privileges.

**Comment by emrahcom** [comment]
Looks like `anonymousdomain` is enabled

**Comment by karthikeyanreactdev** [comment]
from where i  have to remove anonymousdomain ??

**Comment by emrahcom** [comment]
`/etc/jitsi/meet/yourdomain-config.js`

By the way, did you restart `jicofo` after updating?

**Comment by karthikeyanreactdev** [comment]
after removing the anonymousdomain still the issue exists

**Comment by emrahcom** [comment]
Error messages?

**Comment by emrahcom** [comment]
By the way, `token_moderation` has not been updated for a long time, I'm not sure it works with the latest Jitsi releases. You may try [token_affiliation](https://github.com/jitsi-contrib/prosody-plugins/tree/main/token_affiliation) as an alternative.

**Comment by karthikeyanreactdev** [comment]
now working , But members are  able to join without moderator consent

**Comment by karthikeyanreactdev** [comment]
after joining as first user they become moderator. if they joined after moderator they are participants

**Comment by emrahcom** [comment]
In my opinion, check the instructions again. I think you miss a step.

**Comment by karthikeyanreactdev** [comment]
I will check and update
